### ISSUE TITLE
Question related to reproducing result train with cifar 100.

### ISSUE THREAD

**Issue by nesquiq** [scok30/MAE-CIL#3]
Hello,

First of all, thank you for your work and for sharing the code.

I am opening this issue because I encountered some difficulties in reproducing the reported results. Can I simply run the 'sv.sh' script to reproduce the results presented in 'figure.4' and 'table.1' from your ICCV version of the paper? (I am already aware that the reconstruction with mae from masked memory buffer is missing, as mentioned in a previous issue, but I'm hoping for at least a corresponding reproduction result.)

I have noticed a conflict between the description of the loss and the implementation details in your code. The paper mentions the use of a 1024 batch size with AdamW when training from scratch. However, in the 'cifar_dytox_plusplus.yaml' file, it specifies a batch size of 128 per GPU, and 'sv.sh' uses 2 GPUs, resulting in a total of 256 batches. And also lr in the yaml file is set to 5e-4, which differs from the 1e-4 like mentioned in the paper. Is the hyperparameter setup for CIFAR100 different from what you've described in '4.1 Benchmarks and Implementation'?

Your total loss is composed of "L_t = lambda_cls * L_cls + lambda_rec * L_rec + lambda_det * L_det." But if I follow the 'cifar_dytox_plusplus.yaml' file, auto_kd is turned on, and L_kd is also used, with (1 - lambda_kd) being multiplied to L_cls. This makes your loss computed as "L_t  = (1 - lambda_kd) * (L_cls + lambda_rec * L_rec + lambda_det * L_det) + lambda_kd * L_kd" (lambda_cls is not used maybe). Is this consistent with your description in the paper?  

I would greatly appreciate your assistance in resolving these issues.

Thank you.

**Comment by E2E-AD** [comment]
Hello, thank you for your interest in our work. I will respond to your questions one by one.

1.The term "setting" mentioned in the paper refers to the configuration used to standardize settings across various datasets, such as the number of GPUs, learning rates, and so on. During our code organization, we found that the results can be reproduced with just 2 GPUs. In order to reduce the resource requirements for replicating the paper, we have temporarily uploaded one version. We can also provide the training configuration that uses 8 GPUs if needed.

2.Regarding "auto_kd," it is a common configuration in existing continual learning. We did not specifically mention this in the paper because most continual learning models perform inter-task distillation. This is similar to how many recent papers no longer explicitly mention weight decay in the optimizer within the loss function. Additionally, concerning the parameters of the "cls loss," in our implementation, we have maintained the same ratio for the three hyperparameters within the loss function (1:100:100), which is not different from what is described in the paper. 

Finally, we welcome you to explore more training hyperparameters to validate the robustness of our method regarding training parameters.

**Comment by scok30** [comment]
Hello, thank you for your interest in our work. I will respond to your questions one by one.

1.The term "setting" mentioned in the paper refers to the configuration used to standardize settings across various datasets, such as the number of GPUs, learning rates, and so on. During our code organization, we found that the results can be reproduced with just 2 GPUs. In order to reduce the resource requirements for replicating the paper, we have temporarily uploaded one version. We can also provide the training configuration that uses 8 GPUs if needed.

2.Regarding "auto_kd," it is a common configuration in existing continual learning. We did not specifically mention this in the paper because most continual learning models perform inter-task distillation. This is similar to how many recent papers no longer explicitly mention weight decay in the optimizer within the loss function. Additionally, concerning the parameters of the "cls loss," in our implementation, we have maintained the same ratio for the three hyperparameters within the loss function (1:100:100), which is not different from what is described in the paper. 

Finally, we welcome you to explore more training hyperparameters to validate the robustness of our method regarding training parameters.
